src/ophyd_async/epics/core/_aioca.py

```diff
@@ -293,13 +293,29 @@ async def put(self, value: SignalDatatypeT | None, wait: bool):
             write_value = self.initial_values[self.write_pv]
         else:
             write_value = self.converter.write_value(value)
-        await caput(
-            self.write_pv,
-            write_value,
-            datatype=self.converter.write_dbr,
-            wait=wait,
-            timeout=None,
-        )
+        try:
+            await caput(
+                self.write_pv,
+                write_value,
+                datatype=self.converter.write_dbr,
+                wait=wait,
+                timeout=None,
+            )
+        except CANothing as exc:
+            # If we ran into a write error, check to see if there is a list
+            # of valid choices, and if the value we tried to write is in that list.
+            valid_choices = self.converter.metadata.get("choices")
+            if valid_choices is not None and len(valid_choices) > 0:
```

Comment:
Could: 
```suggestion
            if valid_choices:
```
is equivalent.